### PR TITLE
pallet-nft using pallet-fees

### LINKED ISSUE #959: Internal usage of pallet-fees
#### Tasks
- [ ] Update pallet-bridge to use pallet-fees https://github.com/centrifuge/centrifuge-chain/pull/933
- [ ] Update pallet-nft to use pallet-fees

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

Adapt `pallet-nft` in order to use `pallet-fees`.

Fixes #959

## Changes and Descriptions

- `pallet-nft` no longer uses its own mechanism to configure and pay fees.
- `pallet-bridge-mapping` no longer needs `pallet-nft` as dependency.
- `pallet-bridge-mapping` mock simplified.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
```sh
cargo test -p pallet-nft
cargo test -p pallet-bridge-mapping
cargo test --workspace --release --features test-benchmarks,try-runtime
```

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [ ] I have made corresponding changes to the documentation:
  - *Should we notify somehow the new way of configuring fees for this pallet?*
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I rebased on the latest `parachain` branch

### CODE REVIEW

File: pallets/bridge-mapping/src/lib.rs [hunk 1]
```diff
@@ -65,6 +65,15 @@ pub mod pallet {
 		/// Admin is able to set/remove resource mappings.
 		type AdminOrigin: EnsureOrigin<Self::Origin>;
 
+		/// Resource id type
+		type ResourceId: Member
```
Comment (lemunozm) [inline pallets/bridge-mapping/src/lib.rs:69]: Generalizing here, we can avoid the tight coupling with `pallet_nft`, avoiding the waterfall dependency that this implies.

File: pallets/bridge-mapping/Cargo.toml [hunk 2]
```diff
@@ -13,25 +13,14 @@ targets = ['x86_64-unknown-linux-gnu']
 [dependencies]
```
Comment (mikiquantum) [inline pallets/bridge-mapping/Cargo.toml:13]: I think we can get rid of this palle altogether, we currently do not use it and this was meant to be a generic way of supporting resources to actions for a different bridge setup.

File: pallets/nft/src/lib.rs [hunk 3]
```diff
@@ -166,29 +101,16 @@ pub mod pallet {
...
-			+ MaybeSerializeDeserialize
-			+ TypeInfo;
-
 		/// Resource hash id.
 		///
 		/// This type was initially declared in the bridge pallet but was moved here
 		/// to avoid circular dependencies.
 		#[pallet::constant]
-		type HashId: Get<Self::ResourceId>;
+		type HashId: Get<ResourceId>;
```
Comment (mikiquantum) [inline pallets/nft/src/lib.rs:109]: I know this is out of scope of this PR, but do you think we can rename this type to ResourceHashId instead? currently is very un-descriptive.

File: pallets/nft/src/lib.rs [hunk 4]
```diff
@@ -266,15 +188,12 @@ pub mod pallet {
 
 			// Returns a Ethereum-compatible Keccak hash of deposit_address + hash(keccak(name+value+salt)) of each proof provided.
 			let bundled_hash = Self::get_bundled_hash_from_proofs(proofs, deposit_address);
-			Self::deposit_event(Event::<T>::DepositAsset(bundled_hash));
+			Self::deposit_event(Event::<T>::DepositAsset(bundled_hash)); //
```
Comment (mikiquantum) [inline pallets/nft/src/lib.rs:191]: MIssing comment?

**Review by mikiquantum — COMMENTED** [review]
Looks good, just a few clean up comments 👍

File: pallets/nft/src/lib.rs (hunk 3 above)
Comment (lemunozm) [reply pallets/nft/src/lib.rs:109]: I agree!

File: pallets/bridge-mapping/Cargo.toml (hunk 2 above)
Comment (lemunozm) [reply pallets/bridge-mapping/Cargo.toml:13]: Ok, maybe it's better to do it in another PR, do you agree?

Comment (mikiquantum) [reply pallets/bridge-mapping/Cargo.toml:13]: Yep!

**Review by mikiquantum — APPROVED** [review]
👍

**Pull request merged**